src/widgets/textarea/lv_textarea.c

```diff
@@ -116,14 +116,25 @@ void lv_textarea_add_char(lv_obj_t * obj, uint32_t c)
 
     const char * letter_buf = (char *)&u32_buf;
 
+    uint32_t c2 = c;
 #if LV_BIG_ENDIAN_SYSTEM
     if(c != 0) while(*letter_buf == 0) ++letter_buf;
+
+    // The byte order may or may not need to be swapped here to get correct c_uni below,
+    // since lv_textarea_add_text is ordering bytes correctly bfore calling lv_textarea_add_char.
```

Comment:
```suggestion
    // since lv_textarea_add_text is ordering bytes correctly before calling lv_textarea_add_char.
```

Please use `/* */` comments.